### PR TITLE
Added HMRC validations for micro and "other" entities

### PR DESCRIPTION
**Pull request by allenbross-wf**
#### Description of change

Added validations for:
 Micro Companies
Micro LLPs
Other Companies
Other LLPs

#### Steps to Test

CI - conformance suite was updated.

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by derekgengenbacher-wf — DISMISSED** [review]

**Review by austinmatherne-wk — DISMISSED** [review]

File: arelle/plugin/validate/HMRC/ValidateHmrc.py [hunk 1]
```diff
@@ -387,11 +452,14 @@ def _evaluateCode(self, code: str) -> CodeResult:
             result = self._evaluateTextPattern(textValidationPatterns)
         elif code in (CO_AUDIT, LP_AUDIT):
             result = self._evaluateAuditFacts()
-        elif code in (CO_MED_CO, LP_MED_LP):
-            result = self._evaluateMedFact()
+        elif code == CO_DIR_REP:
+            result = self._evaluateDirectorFacts()
+        elif code in (CO_MED_CO, LP_MED_LP, CO_QUAL_AUDIT, LP_QUAL_AUDIT):
+            result = self._evaluateRequiredSingleFact(code)
```
Comment (aaroncameron-wk) [inline arelle/plugin/validate/HMRC/ValidateHmrc.py:458]: Maybe:
```suggestion
        elif code in SINGLE_CONCEPT_EVALUATIONS:
            result = self._evaluateRequiredSingleFact(SINGLE_CONCEPT_EVALUATIONS[code])
```
At least `code in SINGLE_CONCEPT_EVALUATIONS` so we don't have to define the associated codes both here and in `SINGLE_CONCEPT_EVALUATIONS`

File: arelle/plugin/validate/HMRC/ValidateHmrc.py [hunk 2]
```diff
@@ -330,13 +350,30 @@ def _evaluateTextPattern(self, pattern: dict[str, tuple[tuple[str, Pattern[str]]
...
+                    conceptList=missingConcepts,
+                    message="Facts tagged with the DateSigningDirectorsReport and DirectorSigningDirectorsReport must exist with non-nil values. "
+                            "There are no facts tagged with the concepts: %(conceptList)s"
+                )
+        return CodeResult()
+
     def _evaluateAuditFacts(self) -> CodeResult:
         """
-        Logs an error when an audited report does not facts tagged with the concepts of "DateAuditorsReport",  "OpinionAuditorsOnEntity"
+        Logs an error when a director report does not facts tagged with the concepts of "DateSigningDirectorsReport" and  "DirectorSigningDirectorsReport"
```
Comment (aaroncameron-wk) [inline arelle/plugin/validate/HMRC/ValidateHmrc.py:374]: I think this was probably meant for `_evaluateDirectorFacts`.

File: arelle/plugin/validate/HMRC/ValidateHmrc.py (hunk 1 above)
Comment (aaroncameron-wk) [reply arelle/plugin/validate/HMRC/ValidateHmrc.py:458]: Pre-existing, but you could update the `TEXT_VALIDATION_PATTERNS` check above with this same pattern.

**Review by austinmatherne-wk — APPROVED** [review]

**Pull request merged**